src/Uno.Extensions.Navigation.UI/Navigator.cs

```diff
@@ -541,7 +541,8 @@ protected virtual async Task<bool> RegionCanNavigate(Route route, RouteInfo? rou
 			var currentRouteMap = Resolver.FindByPath(Route.Base);
 			if (currentRouteMap != null || routeMap != null)
 			{
-				if (currentRouteMap?.Parent != routeMap?.Parent)
+				if (routeMap?.Parent is not null &&
+					currentRouteMap?.Parent != routeMap?.Parent)
```

Comment:
```suggestion
					currentRouteMap?.Parent != routeMap.Parent)
```
you just did a null check there is no reason to safeguard this from routeMap being potentially null here...